src/snapred/backend/data/LocalDataService.py

```diff
@@ -303,230 +253,85 @@ def _constructReductionDataRoot(self, runNumber: str, useLiteMode: bool) -> Path
         mode = "lite" if useLiteMode else "native"
         return reductionStateRoot / mode / runNumber
 
-    @validate_call
-    def _constructReductionDataPath(self, runNumber: str, useLiteMode: bool, version: Version) -> Path:
-        return self._appendVersion(self._constructReductionDataRoot(runNumber, useLiteMode), version)
-
-    @validate_call
-    def _constructReductionRecordFilePath(self, runNumber: str, useLiteMode: bool, version: Version) -> Path:
-        recordPath = self._constructReductionDataPath(runNumber, useLiteMode, version) / "ReductionRecord.json"
-        return recordPath
-
     @validate_call
     def _constructReductionDataFilePath(self, runNumber: str, useLiteMode: bool, version: Version) -> Path:
         stateId, _ = self._generateStateId(runNumber)
         fileName = wng.reductionOutputGroup().stateId(stateId).version(version).build()
         fileName += Config["nexus.file.extension"]
-        filePath = self._constructReductionDataPath(runNumber, useLiteMode, version) / fileName
+        filePath = self.reductionIndexor(runNumber, useLiteMode).versionPath(version) / fileName
         return filePath
 
-    ##### INDEX / VERSION METHODS #####
+    ##### VERSIONING / INDEXING METHODS #####
+
+    def _statePathForWorkflow(self, stateId: str, useLiteMode: bool, indexorType: IndexorType):
+        """
+        NOTE for reduction, use the runNumber in place of the stateId
+        """
+        match indexorType:
+            case IndexorType.CALIBRATION:
+                path = self._constructCalibrationStatePath(stateId, useLiteMode)
+            case IndexorType.NORMALIZATION:
+                path = self._constructNormalizationStatePath(stateId, useLiteMode)
+            case IndexorType.REDUCTION:
+                # NOTE the "stateId" should actually be a runNumber for reduction
+                path = self._constructReductionDataRoot(stateId, useLiteMode)
+            case _:
+                mode = "lite" if useLiteMode else "native"
+                path = self._constructCalibrationStateRoot(stateId) / mode
+        return path
```

Comment:
An `Indexor` must be initialized with a directory, and it will manage the files inside that directory.  This method produces the needed directory for calibration, normalization, and reduction `Indexor`s.

Reply:
Yes, `groupingMap` doesn't make reference to this method